### PR TITLE
User creation closes  #10 #65 #

### LINKED ISSUE #10: As an unauthenticated user, when I create an account, I can request that my account become a borrower

### PR DESCRIPTION
**Pull request by mikedao**
H'okay, folks.

Modified the Tenants controller to allow creation and association with a user after the user is created. Also refactored it and pulled out some logic to a private method.

Modified Users controller to reflect the new order in user signup/creation.

Nav bar was changed to stop linking to the static choose pages controller.

New user form now has a checkbox to indicate if they are a borrower.

Removed irrelevant tests, modified existing tests.

Removed static pages controller, views and tests.

### CODE REVIEW

File: app/controllers/tenants_controller.rb [hunk 1]
```diff
@@ -9,26 +9,33 @@ def new
...
+      flash[:notice] = "Please try again."
+      redirect_to new_tenant_path
+    end
+  end
+
   def org_params
     params.require(:tenant_signup).permit(:location, :organization)
   end
+
+  def org_check
```
Comment (EmilyMB) [inline app/controllers/tenants_controller.rb]: Beautiful. Strong params are like :sunny: and :rainbow: and :tulip:

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -41,7 +41,11 @@ def create_user(user)
     session[:user_id] = user.id
     send_welcome_email(user)
     flash[:notice] = "Thank you for creating an account."
-    redirect_to root_path
+    if tenant?
```
Comment (EmilyMB) [inline app/controllers/users_controller.rb]: Maybe change this to if borrower?

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -49,6 +53,10 @@ def send_welcome_email(user)
     UserMailer.welcome_lender(user).deliver_now if user.lender?
   end
 
+  def tenant?
```
Comment (EmilyMB) [inline app/controllers/users_controller.rb]: Borrower?

File: app/views/layouts/_nav_bar.html.erb [hunk 4]
```diff
@@ -16,7 +16,7 @@
     <%= menu_item "Pending Loans", pending_loan_path %>
     <%= menu_item "Admin Dashboard", admin_dashboard_path if current_user && current_user.admin? %>
     <% if current_user.nil? %>
-    	<%= menu_item "Signup", choose_path %>
+    	<%= menu_item "Signup", new_user_path %>
```
Comment (EmilyMB) [inline app/views/layouts/_nav_bar.html.erb]: This feels so much more RESTful :smile: :diamonds:

File: app/views/static_pages/choose.html.erb [hunk 5]
```diff
@@ -1,5 +0,0 @@
-<h1>Are You A...</h1>
-<center>
-<h2><%= link_to "Borrower", new_tenant_path %></h2>
-<h2><%= link_to "Lender", new_user_path %></h2>
-</center>
```
Comment (EmilyMB) [inline app/views/static_pages/choose.html.erb]: Won't miss this!

File: config/environments/development.rb [hunk 6]
```diff
@@ -7,7 +7,7 @@
   config.consider_all_requests_local       = true
   config.action_controller.perform_caching = false
 
-  config.action_mailer.perform_deliveries = true
+  config.action_mailer.perform_deliveries = false
```
Comment (EmilyMB) [inline config/environments/development.rb]: No more emails? :email:

File: test/integration/user_signup_test.rb [hunk 7]
```diff
@@ -103,22 +95,20 @@ class CreateAUserTest < ActionDispatch::IntegrationTest
...
     fill_in "signup[email]", with: "email@example.com"
+    check "signup[tenant]"
     click_link_or_button "Create Account"
+    fill_in "tenant_signup[location]", with: "location"
+    fill_in "tenant_signup[organization]", with: "organization"
+    click_link_or_button "Create Organization"
 
     assert_equal root_path, current_path
-    assert page.has_content?("Thank you for creating an account.")
+    assert page.has_content?("Organization Created.")
```
Comment (EmilyMB) [inline test/integration/user_signup_test.rb]: Maybe assert that a more descriptive message is displayed? Maybe "Your organization, #{org}  has been successfully created. You will receive an email at #{user.email} when your org has been approved to request loans"

File: config/environments/development.rb (hunk 6 above)
Comment (mikedao) [inline config/environments/development.rb]: Just in development. Still activated in prod.

File: test/integration/user_signup_test.rb (hunk 7 above)
Comment (mikedao) [inline test/integration/user_signup_test.rb]: This is a flash message, should it be something long?

Comment (EmilyMB) [inline test/integration/user_signup_test.rb]: Fair enough, I guess we can put all the explanation in their welcome email.

**Pull request merged**